akka-stream-typed/src/main/scala/akka/stream/typed/javadsl/ActorFlow.scala

```diff
@@ -106,4 +120,18 @@ object ActorFlow {
       .ask[I, Q, A](parallelism)(ref)((i, ref) => makeMessage(i, ref))(timeout.toMillis.millis)
       .asJava
 
+  /**
+   * Use for messages whose response is known to be a [[akka.pattern.StatusReply]]. When a [[akka.pattern.StatusReply#success]] response
+   * arrives the future is completed with the wrapped value, if a [[akka.pattern.StatusReply#error]] arrives the future is instead
+   * failed.
+   */
+  def askWithStatus[I, Q, A](
+      parallelism: Int,
+      ref: ActorRef[Q],
+      timeout: java.time.Duration,
+      makeMessage: (I, ActorRef[StatusReply[A]]) => Q): Flow[I, A, NotUsed] =
```

Comment:
This needs to be a Java funciton type, I think `java.util.Function` is good. (Looks like an oversight in the pre-existing ask-methods here, but we can't fix those because of bincomp).